build/DownloadCFamilyPlugin/DownloadAndExtractSonarJS.cs

```diff
@@ -0,0 +1,119 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace DownloadCFamilyPlugin
+{
+    /// <summary>
+    /// Downloads the Sonar JavaScript/TypeScript jar and extracts the files that need to be
+    /// embedded in the SLVS vsix
+    /// </summary>
+    /// <remarks>
+    /// Assumptions:
+    /// * we are downloading the sonar-js plugin jar e.g. https://binaries.sonarsource.com/Distribution/sonar-javascript-plugin/sonar-javascript-plugin-6.2.0.12043.jar
+    /// * the jar contains a .tgz archive containing ESLintBridge (a NodeJS app)
+    /// * the jar contains various .json files, some with well-known names.
+    /// 
+    /// We want the task to fail if it cannot locate all of the expected files (otherwise we'll build an invalid VSIX).
+    /// 
+    /// Downloading and extracting the files can be slow so we want to skip those steps if possible.
+    /// </remarks>
+    public class DownloadAndExtractSonarJS: Task
```

Comment:
Unrelated to this PR: I just noticed that the project, sln and namespace mention only CFamily. I think it's worth renaming it in the future.

Reply:
That's one of the sub-items in the ticket: #2246